### PR TITLE
Fold maxRevision into constructHistory

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
For DaemonSetsController#constructHistory, it calls maxRevision to get the max revision of old histories.
This PR folds maxRevision into constructHistory.

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/ok-to-test
/kind cleanup
/priority backlog

**Comment by tedyu** [comment]
ping @mikedanese for review

**Comment by tedyu** [comment]
ping @cheftako for review

Thanks

File: pkg/controller/daemon/update.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ func (dsc *DaemonSetsController) constructHistory(ds *apps.DaemonSet) (cur *apps
 	if err != nil {
 		return nil, nil, err
 	}
+	max := int64(0)
```
Comment (mikedanese) [inline pkg/controller/daemon/update.go]: why int64?

Comment (tedyu) [reply pkg/controller/daemon/update.go]: No particular reason - just to be on the safe side.

File: pkg/controller/daemon/update.go [hunk 2]
```diff
@@ -187,17 +191,6 @@ func (dsc *DaemonSetsController) cleanupHistory(ds *apps.DaemonSet, old []*apps.
 	return nil
 }
 
-// maxRevision returns the max revision number of the given list of histories
-func maxRevision(histories []*apps.ControllerRevision) int64 {
```
Comment (mikedanese) [inline pkg/controller/daemon/update.go]: I don't mind removing this function, but I would leave it in a separate loop vs combing it with the above loop. There's a lot going on in that loop already, and I don't think this change helps with the clarity of what's going on.

**Comment by mikedanese** [comment]
What motivates this change?

**Comment by tedyu** [comment]
I was reading this code and thought it is not efficient making another loop just to find the max where the max is readily available by the first loop.

**Comment by mikedanese** [comment]
In general, we should avoid optimization at the expense of readability without quantitative evidence that the code is a bottleneck in one of the kube-* processes. We should write code that is easy to maintain and trust the compiler until it becomes a problem. Specifically with loop optimizations, it can't be generalized that combined loops are faster than distributed loops. It depends on what how the loop is using cpu caches.

**Comment by tedyu** [comment]
Can reopen if bottleneck is observed in this area.

**Pull request closed without merging**